src/app/src/mainwindow.cpp

```diff
@@ -230,19 +233,54 @@ void MainWindow::on_ledIntensityEdit_valueChanged(double value) {
 }
 
 
+/*
+ * Check if drive on windows has sufficient space for acquisition.
+ *
+ * @param drive letter selected in config file
+ * @param fps setting of acquisition
+ * @param duration of acquisition
+ * @param TODO
+ *
+ * @returns boolean true if space is available
+*/
+bool available_space_in_drive(std::filesystem::path driver_name, double fps,double duration,double frameBytes){
+    ULARGE_INTEGER  lpTotalNumberOfFreeBytes = { 0 };
+    GetDiskFreeSpaceEx(
+        driver_name.c_str(),
+        nullptr,
+        nullptr,
+        & lpTotalNumberOfFreeBytes
+        );
+    std::stringstream space_string,driver_string;
+    space_string << lpTotalNumberOfFreeBytes.QuadPart;
+    driver_string << driver_name;
+    spdlog::info("Drive {} has: {} bytes free for acquisition",driver_string.str(),space_string.str());
+    return lpTotalNumberOfFreeBytes.QuadPart > fps * duration * frameBytes;
+}
+
+
 /*
  * Frame Rate edit box slot, called when users changes the FPS value.
  *
  * @param value The updated FPS value.
  */
 void MainWindow::on_frameRateEdit_valueChanged(double value) {
+    uns32 Framebytes =  m_camera->GetFrameBytes();
```

Comment:
this is nit-picky, but can you use `framebytes` or `frameBytes` instead of `FrameBytes`, using a capital letter to start usually indicates a class name or a public function name, it's not enforced by the compile but it's a typical style of c++ code